Microsoft.Toolkit.Uwp.UI.Controls/MosaicControl/MosaicControl.cs

```diff
@@ -0,0 +1,1105 @@
+                count = xamlChildren.Count;
+            }
+
+            // instanciate all elements not created yet
+            for (int x = 0; x < numberSpriteToInstanciate - count; x++)
+            {
+                if (strategy == UIStrategy.Composition)
+                {
+                    var sprite = this.containerVisual.Compositor.CreateSpriteVisual();
+                    this.containerVisual.Children.InsertAtTop(sprite);
+                    compositionChildren.Add(sprite);
+                }
+                else
+                {
+                    var rectangle = new Rectangle();
+                    this.containerElement.Children.Add(rectangle);
+                    xamlChildren.Add(rectangle);
+                }
+            }
+
+            // remove elements not used now
+            for (int x = 0; x < count - numberSpriteToInstanciate; x++)
+            {
+                if (strategy == UIStrategy.Composition)
+                {
+                    var element = this.containerVisual.Children.FirstOrDefault() as SpriteVisual;
+                    this.containerVisual.Children.Remove(element);
+                    compositionChildren.Remove(element);
+                }
+                else
+                {
+                    var element = this.containerElement.Children.FirstOrDefault() as Rectangle;
+                    this.containerElement.Children.Remove(element);
+                    xamlChildren.Remove(element);
+                }
+            }
+
+            // Change positions+brush for all actives elements
+            for (int y = 0; y < numberImagePerRow; y++)
+            {
+                for (int x = 0; x < numberImagePerColumn; x++)
+                {
+                    int index = (y * numberImagePerColumn) + x;
+
+                    if (strategy == UIStrategy.Composition)
+                    {
+                        var sprite = compositionChildren[index];
+                        sprite.Brush = this.brushVisual;
+                        sprite.Offset = new Vector3((float)((x * imageWidth) + offsetVerticalAlignment), (float)((y * imageHeight) + offsetHorizontalAlignment), 0);
+                        sprite.Size = new Vector2((float)imageWidth, (float)imageHeight);
+                    }
+                    else
+                    {
+                        var rectangle = xamlChildren[index];
+                        rectangle.Fill = this.brushXaml;
+
+                        Canvas.SetLeft(rectangle, (x * imageWidth) + offsetVerticalAlignment);
+                        Canvas.SetTop(rectangle, (y * imageHeight) + offsetHorizontalAlignment);
+                        rectangle.Width = imageWidth;
+                        rectangle.Height = imageHeight;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Gets or sets an X offset of the image
+        /// </summary>
+        public double OffsetX
```

Comment:
OffsetX and OffsetY are not destined to manage the layout but the starting position of the repeated image. It's like properties AlignmentX and AlignmentY of the ImageBrush class in WPF.